File: Beecrowd/Python/1040.py
from decimal import Decimal, getcontext

getcontext().prec = 10

notas = list(map(Decimal, input().split()))
pesos = [2, 3, 4, 1]
media = Decimal('0')

for i, nota in enumerate(notas):
    nota_real = nota * Decimal(str((pesos[i]/10)))
    media += nota_real
    print(media)
print(f'Media: {media:.1f}')

if media >= 7:
    print('Aluno aprovado.')
elif media < 5:
    print('Aluno reprovado.')
elif 5 <= media < 7:
    print('Aluno em exame.')
    nota_exame = Decimal(input())
    media_final = (nota_exame + media)/2
    
    print(f'Nota do exame: {nota_exame:.1f}')
    if media_final >= 5:
        print('Aluno aprovado.')
    elif media_final < 5:
        print('Aluno reprovado.')
    print(f'Media final: {media_final:.1f}')

File: Beecrowd/C/1040.c
#include <stdio.h>

int main(void) {
    int pesos[5] = {2, 3, 4, 1};
    float notas[5];
    float media = 0;

    for(int i = 0; i < 4; i++) {
        scanf("%f", &notas[i]);
        notas[i] *= pesos[i];
        media += notas[i];
    }

    printf("Media: %.1f\n", media / 10);

    if (media >= 70) {
        printf("Aluno aprovado.\n");
    } else if (media >= 50) {
        printf("Aluno em exame.\n");
        printf("Nota do exame: ");

        float exame;
        scanf("%f", &exame);
        printf("%.1f\n", exame);
        media = (media + exame*10) / 2;

        if (media >= 50) {
            printf("Aluno aprovado.\n");
        } else {
            printf("Aluno reprovado.\n");
        }
        printf("Media final: %.1f\n", media / 10);
        
    } else {
        printf("Aluno reprovado.\n");
    }

    return 0;
}